### DISCUSSION TITLE
Passthrough is not supported, GL is swiftshader

### DISCUSSION THREAD

**Discussion by schandra09net** [cefsharp/CefSharp#3824] [General]
Hi @amaitland,

I am using CefSharp 91.1.211 in my C# WPF application(Windows 10), the chromiumbrowser is used inside grids.

Xaml definition:
<bll:BaseStageItem>
	<Grid>
		<Grid x:Name="MainGrid">
			<Grid.RowDefinitions>
				<RowDefinition Height="0" x:Name="AddressBarRowDef" />
				<RowDefinition Height="*" />
			</Grid.RowDefinitions>

			<Grid Grid.Row="1">
				<Grid x:Name="Browser" Background="LightGray" ClipToBounds="True" >
					<Grid x:Name="SnapshotRegionGrid">
						<cefSharp:ChromiumWebBrowser  
								Name="ViewControl" 
								 FocusVisualStyle="{x:Null}"                                                                                                         
								 Focusable="True"
								 ScrollViewer.HorizontalScrollBarVisibility="Auto"
								 ScrollViewer.VerticalScrollBarVisibility="Auto"
								 VerticalAlignment="Stretch"
								 HorizontalAlignment="Stretch"
								 Address="http://www.facebook.com"/>
					</Grid>
				</Grid>
			</Grid>
		</Grid>
	</Grid>
<bll:BaseStageItem>

and initialization settings:

CefSettings cefSettings = new CefSettings();
string assemblyFolder = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
//cefSettings.BrowserSubprocessPath = Path.Combine(assemblyFolder, "CefSharp.BrowserSubprocess.exe");
cefSettings.CachePath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "CefSharp\\Cache");
cefSettings.LogSeverity = LogSeverity.Warning;
cefSettings.LogFile = Path.Combine(assemblyFolder, "cefbrowser.log");
cefSettings.SetOffScreenRenderingBestPerformanceArgs();
cefSettings.MultiThreadedMessageLoop = true;
cefSettings.CefCommandLineArgs.Add("enable-gpu");
cefSettings.CefCommandLineArgs.Add("off-screen-rendering-enabled");        
Cef.Initialize(cefSettings);

After running the WPF app, the browser loads and navigates to www.facebook.com", and when i type text in username field, then 
as soon as i press keyboard key, the browser is freezing (not usable/clickable, until reload the page), and in logs i see

[0924/110541.566:ERROR:viz_main_impl.cc(160)] Exiting GPU process due to errors during initialization
[0924/110541.735:ERROR:gpu_init.cc(440)] Passthrough is not supported, GL is swiftshader

"WerFault.exe" is getting created and stopped in TaskManager.

Can you guys help me, i am not able to type in the browser.

You help is so much appreciated.

**Comment by amaitland** [comment]
In future please use https://github.com/cefsharp/CefSharp/discussions to ask questions. 

Test with https://github.com/cefsharp/CefSharp.MinimalExample and confirm the problem reproduces. 

> WPF the browser stops redrawing when running on Intel Iris Xe Integrated GPU(11th Gen). Intel have released an updated driver, see https://www.intel.com/content/www/us/en/support/articles/000058136/graphics/graphics-for-11th-generation-intel-processors.html See dotnet/wpf#3817 for workaround if you are unable to install the updated driver.

As per https://github.com/cefsharp/CefSharp/releases/tag/v93.1.111 if you are using an Intel 11th Gen iris graphics cards then you'll need to update your driver.

**Reply by schandra09net** [reply]
Thanks @amaitland.
